selfdrive/boardd/boardd.cc

```diff
@@ -428,18 +400,35 @@ void hardware_control_thread() {
   uint16_t prev_fan_speed = 999;
   uint16_t ir_pwr = 0;
   uint16_t prev_ir_pwr = 999;
+  bool prev_charging_disabled = false;
   unsigned int cnt = 0;
 
   while (!do_exit && panda->connected) {
     cnt++;
     sm.update(1000); // TODO: what happens if EINTR is sent while in sm.update?
 
     if (sm.updated("thermal")){
+      // Fan speed
       uint16_t fan_speed = sm["thermal"].getThermal().getFanSpeed();
       if (fan_speed != prev_fan_speed || cnt % 100 == 0){
         panda->set_fan_speed(fan_speed);
         prev_fan_speed = fan_speed;
       }
+
+      // Charging mode
+      bool charging_disabled = sm["thermal"].getThermal().getChargingDisabled();
+      if (charging_disabled != prev_charging_disabled){
+#ifdef QCOM
```

Comment:
This seems confusing. Why isn't the whole thing gated on QCOM?

Reply:
It was at first, but then it complains about prev_charging_disabled being unused in the non-qcom case

Reply:
I'd gate prev_charging_disabled on QCOM too then. We could also define an UNUSED macro like we do in panda